module/epochs/machine_account.go

```diff
@@ -45,9 +55,107 @@ func init() {
 	}
 }
 
+// MachineAccountConfigValidator is used to validate that a machine account is
+// configured correctly.
+type MachineAccountConfigValidator struct {
+	unit   *engine.Unit
+	log    zerolog.Logger
+	client *client.Client
+	role   flow.Role
+	info   bootstrap.NodeMachineAccountInfo
+}
+
+func NewMachineAccountConfigValidator(
+	log zerolog.Logger,
+	flowClient *client.Client,
+	role flow.Role,
+	info bootstrap.NodeMachineAccountInfo,
+) (*MachineAccountConfigValidator, error) {
+	validator := &MachineAccountConfigValidator{
+		unit:   engine.NewUnit(),
+		log:    log.With().Str("component", "machine_account_config_validator").Logger(),
+		client: flowClient,
+		role:   role,
+		info:   info,
+	}
+	return validator, nil
+}
+
+// Ready will launch the validator function in a goroutine.
+func (validator *MachineAccountConfigValidator) Ready() <-chan struct{} {
+	return validator.unit.Ready(func() {
+		validator.unit.Launch(func() {
+			validator.validateMachineAccountConfig(validator.unit.Ctx())
+		})
+	})
+}
+
+// Done will cancel the context of the unit, which will end the validator
+// goroutine, if it is still running.
+func (validator *MachineAccountConfigValidator) Done() <-chan struct{} {
+	return validator.unit.Done()
+}
+
+// validateMachineAccountConfig checks that the machine account in use by this
+// BaseClient object is correctly configured. If the machine account is critically
+// mis-configured, or a correct configuration cannot be confirmed, this function
+// will perpetually log errors indicating the problem.
+//
+// This function should be invoked as a goroutine by using Ready and Done.
+//
+func (validator *MachineAccountConfigValidator) validateMachineAccountConfig(ctx context.Context) {
+
+	log := validator.log
+
+	expRetry, err := retry.NewExponential(checkMachineAccountRetryBase)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to create machine account check retry")
+	}
+	backoff := retry.WithJitterPercent(
+		5, // 5% jitter
+		retry.WithMaxDuration(checkMachineAccountRetryMax, expRetry),
+	)
+
+	err = retry.Do(ctx, backoff, func(ctx context.Context) error {
+		account, err := validator.client.GetAccount(ctx, validator.info.SDKAddress())
+		if err != nil {
+			// we cannot validate a correct configuration - log an error and try again
+			log.Error().
+				Err(err).
+				Str("machine_account_address", validator.info.Address).
+				Msg("failed to validate machine account config - could not get machine account")
+			return retry.RetryableError(err)
+		}
+
+		err = CheckMachineAccountInfo(log, validator.role, validator.info, account)
+		if err != nil {
+			// either we cannot validate the configuration or there is a critical
+			// misconfiguration - log a warning and retry - we will continue checking
+			// and logging until the problem is resolved
+			log.Error().
+				Err(err).
+				Msg("critical machine account misconfiguration")
+			return retry.RetryableError(err)
```

Comment:
Do we need to retry if checking the machine info fails ? What will change about the info between retries ?

Reply:
Some failures can be resolved out-of-band and would be addressed after retrying. For example, a temporary failure to connect to the access node, or an insufficient balance. 

For the failures that can't be resolved without a restart, if we aren't going to crash, it's better to periodically log so that it's easier to find this problem (eg. a query for this log over the last hour would still find it).